tools/cli/commands/rsync.js

```diff
@@ -68,17 +70,95 @@ export async function rsyncInit( argv ) {
 	}
 
 	if ( argv.watch ) {
-		console.log( 'jetpack rsync --watch is now watching for changes to:', sourcePluginPath );
-		const debouncedRsyncToDest = pDebounce(
-			() => rsyncToDest( sourcePluginPath, finalDest, argv.dest ),
-			10000 // Todo: at 10000ms, standard Jetpack build ran this twice for me, adjust further?
-		);
-		// eslint-disable-next-line no-unused-vars
-		chokidar.watch( sourcePluginPath ).on( 'change', async changedPath => {
-			await debouncedRsyncToDest();
-		} );
+		let watcher;
+		const rsyncAndUpdateWatches = async ( event, eventfile ) => {
+			if ( argv.v ) {
+				console.debug( `rsync due to event ${ event } for ${ eventfile }` );
+			}
+
+			const paths = await rsyncToDest( sourcePluginPath, finalDest );
+
+			if ( ! watcher ) {
+				watcher = chokidar.watch( [], {
+					cwd: sourcePluginPath,
+					followSymlinks: false,
+					disableGlobbing: true,
+					ignoreInitial: true,
+					depth: 0,
+				} );
+
+				// Always watch the plugin base dir.
+				watcher.add( '.' );
+
+				// Also watch the git index for changes to catch `git add`, as that may change which files are synced.
+				watcher.add( path.join( process.cwd(), '.git/index' ) );
+
+				// Watch `.gitignore` and `.gitattributes` in parent dirs, as they too may change which files are synced.
+				// Here we assume sourcePluginPath is always `projects/plugins/whatever`
+				watcher.add( '../.gitignore' );
+				watcher.add( '../.gitattributes' );
+				watcher.add( '../../.gitignore' );
+				watcher.add( '../../.gitattributes' );
+				watcher.add( '../../../.gitignore' );
+				watcher.add( '../../../.gitattributes' );
+
+				watcher.once( 'ready', () => {
+					console.log( 'jetpack rsync --watch is now watching for changes to:', sourcePluginPath );
+					watcher.on( 'all', pDebounce( rsyncAndUpdateWatches, 1000 ) );
```

Comment:
If I run `jetpack build` then rysnc will occur frequently with the `1000` timer. What would be the downside of setting this a bit higher?

Reply:
It's a balance between someone working with JS and having to wait for a full build, and someone editing PHP files and not wanting to wait 10 seconds for the sync, I suppose.

And what if the JS-editing person is using `jetpack watch plugins/jetpack`? Those rebuilds tend to be a lot faster too.